src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition.java

```diff
@@ -113,7 +112,9 @@ public CpsFlowExecution create(FlowExecutionOwner owner, TaskListener listener,
                 return ((CpsFlowFactoryAction2) a).create(this, owner, actions);
             }
         }
-        return new CpsFlowExecution(sandbox ? script : ScriptApproval.get().using(script, GroovyLanguage.get()), sandbox, owner, getDurabilityHint());
+        Queue.Executable exec = owner.getExecutable();
+        FlowDurabilityHint hint = (exec instanceof Run) ? DurabilityHintProvider.suggestedFor(((Run)exec).getParent()) : GlobalDefaultFlowDurabilityLevel.getDefaultDurabilityHint();
```

Comment:
`DurabilityHintProvider` should be operating on `FlowExecutionOwner`, not `Run`. As written, it will be unusable for (say) a `/script` console-like Pipeline runner.

Reply:
AFAIK no such thing exists, nor have I heard of plans to provide one?

Reply:
Does not exist currently, but all the APIs are designed with the intent of making it possible to provide such an engine.